### DISCUSSION TITLE
Reduce the number of cluster (n_process)

### DISCUSSION THREAD

**Discussion by zhouyi0812** [flatironinstitute/CaImAn#1256] [Q&A]
Hello!
I wonder if reduce the number of cluster will speed up the algorithm, what is the disadvantage of it? 
And how to know what is the best number to choose that balance the quality and speed?
Thank you!

**Comment by EricThomson** [comment]
There is a new section in the revised CNMF notebook (the section is called **Optimizing Performance**) that will be in the next release due out in the next day or so:

> If you hit memory issues, there are a few things you can do. First, you may want to lower the number of processors you are using. Each processor uses more RAM, and on a workstation with many processors, you can sometimes get better performance by reducing num_processors_to_use. The best way to determine the optimal number is by trial and error. When you set num_processors_to_use variable to None, it defaults to one less than the total number of CPU cores available (the reason we don't automatically set it to the total number of cores is because in practice this typically leads to worse performance). 

There is no easy answer here, it is system dependent so you have to play with this on your particular setup if you are not getting good results.
